packages/ccip-svm/src/encoders/evmExtraArgsV2.ts

```diff
@@ -0,0 +1,76 @@
+import { encodeAbiParameters, parseAbiParameters, decodeAbiParameters } from 'viem'
+
+/**
+ * EVM Extra Args V2 tag: bytes4(keccak256("CCIP EVMExtraArgsV2"))
+ */
+export const EVM_EXTRA_ARGS_V2_TAG = 0x181dcf10
+
+export interface EVMExtraArgsV2 {
+    gasLimit: bigint
+    allowOutOfOrderExecution?: boolean
+}
+
+/**
+ * Creates properly encoded extraArgs buffer for EVM destinations
+ * @param gasLimit - Gas limit for execution on destination chain (use 0 for token-only transfers)
+ * @param allowOutOfOrderExecution - Whether messages can be executed out of order (default: true)
+ * @returns Properly encoded extraArgs as Uint8Array
+ */
+export function encodeEVMExtraArgsV2({
+    gasLimit,
+    allowOutOfOrderExecution = true
+}: EVMExtraArgsV2): Uint8Array {
+    if (allowOutOfOrderExecution === false) {
+        console.warn(
+            `Warning: Setting allowOutOfOrderExecution to false is not recommended for EVM destinations.`
```

Comment:
recommend we link to docs in this warning:
https://docs.chain.link/ccip/concepts/best-practices/evm#best-practices

and then update test to ensure that some ref to docs remains as that is helpful /educational devX